libs/sync/src/lib/sync/new-sync.service.ts

```diff
@@ -211,53 +212,44 @@ export class NewSyncService {
     sourceId: string,
     source: Source,
     kbId: string, // TO BE REMOVED (useless, kept for compatibility with old code)
-    localBackend?: string,
+    localBackend?: string, // TO BE REMOVED (useless, kept for compatibility with old code)
     labels?: Classification[],
   ): Observable<void> {
-    if (localBackend) {
-      source = {
-        ...source,
-        labels,
-        kb: { standalone: true, zone: '', backend: localBackend, account: 'local', knowledgeBox: kbId },
-      };
-      return this.setSourceData(sourceId, source);
-    } else {
-      return this.sdk.currentKb.pipe(
-        take(1),
-        switchMap((kb) => {
-          if (localBackend) {
-            return of({
+    return this.sdk.currentKb.pipe(
+      take(1),
+      switchMap((kb) => {
+        if (this.sdk.nuclia.options.standalone) {
+          return of({
+            ...source,
+            labels,
+            kb: { ...this.sdk.nuclia.options, knowledgeBox: kb.id },
+          });
+        } else if (source.kb && source.kb.knowledgeBox === kb.id && source.kb.apiKey) {
+          return of(source);
+        } else {
+          return this.getNucliaKey(kb).pipe(
+            map((data) => ({
               ...source,
               labels,
-              kb: { standalone: true, zone: '', backend: localBackend, account: 'local', knowledgeBox: kb.id },
-            });
-          } else if (source.kb && source.kb.knowledgeBox === kb.id && source.kb.apiKey) {
-            return of(source);
-          } else {
-            return this.getNucliaKey(kb).pipe(
-              map((data) => ({
-                ...source,
-                labels,
-                kb: {
-                  zone: this.sdk.nuclia.options.zone,
-                  backend: this.sdk.nuclia.options.backend,
-                  knowledgeBox: data.kbid,
-                  apiKey: data.token,
-                } as NucliaOptions,
-              })),
-            );
-          }
-        }),
-        switchMap((source) => this.setSourceData(sourceId, source)),
-      );
-    }
+              kb: {
+                zone: this.sdk.nuclia.options.zone,
+                backend: this.sdk.nuclia.options.backend,
+                knowledgeBox: data.kbid,
+                apiKey: data.token,
+              } as NucliaOptions,
+            })),
+          );
+        }
+      }),
+      switchMap((source) => this.setSourceData(sourceId, source)),
```

Comment:
when running `setSourceData`, the sync agent returns `Invalid format for kb: Error: zone: Required, apiKey: Required`, Is this expected ?

Reply:
I forgot to push my sync-agent PR :)